File: 4229.Trim Trailing Vowels/solution.cpp
class Solution {
public:
    string trimTrailingVowels(string s) {
        int n = s.size();
        int i = n-1;
        while(i>=0 && isvowel(s[i])){
            i--;
        }
        if(i<0){
            return "";
        }
        return s.substr(0,i+1);
    }

    bool isvowel(char c){
        if(c=='a'||c=='e'||c=='i'||c=='o'||c=='u'){
            return true;
        }
        return false;
    }
};

File: 4229.Trim Trailing Vowels/solution.py
class Solution(object):
    def trimTrailingVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        # n = len(s)
        # h = n-1
        vow = {'a','e','i','o','u'}
        while(s[len(s)-1] in vow):
            s = s[:len(s)-1]
            if(s==""):
                return ""
        return s
        # while(h>=0 and s[h] in vow):
            # h-=1
        # return s[:h+1]